configure.py

```diff
@@ -276,6 +276,15 @@ def src_info(self, typ):
         else:
             raise InternalError("Unknown src info type '%s'" % (typ))
 
+ACCEPTABLE_BUILD_TARGETS = ["static", "shared", "cli", "tests", "bogo_shim"]
+
+def read_build_targets(options):
```

Comment:
The code here should happen once, in `canonicalize_build_targets`, and then stores the result in `options.build_targets`.